src/osd/OSDMap.cc

```diff
@@ -1022,14 +1031,17 @@ int OSDMap::calc_num_osds()
   return num_osd;
 }
 
-void OSDMap::count_full_nearfull_osds(int *full, int *nearfull) const
+void OSDMap::count_full_nearfull_osds(int *full, int *backfill, int *nearfull) const
 {
   *full = 0;
+  *backfill = 0;
   *nearfull = 0;
   for (int i = 0; i < max_osd; ++i) {
     if (exists(i) && is_up(i) && is_in(i)) {
       if (osd_state[i] & CEPH_OSD_FULL)
 	++(*full);
+      else if (osd_state[i] & CEPH_OSD_BACKFILLFULL)
```

Comment:
are these states mutual exclusive? or we just want to ensure that the total number of full / backfull / nearfull less than the total number of osd?